examples/advanced/docker/nvflare_with_docker.ipynb

```diff
@@ -0,0 +1,429 @@
+   "id": "0653cbf2-92f2-4a22-8317-69cfb0266e92",
+   "metadata": {},
+   "source": [
+    "### NVIDIA FLARE with Docker\n",
+    "This notebook shows how to use [NVIDIA FLARE](https://nvflare.readthedocs.io/en/main/index.html) with [Docker](https://www.docker.com/).\n",
+    "\n",
+    "Please make sure you set up a virtual environment and install JupyterLab following the [example root readme](../../README.md).\n",
+    "\n",
+    "Also, make sure that you have cloned the [NVFlare](https://github.com/NVIDIA/NVFlare) repository so you have the source code for building Docker images with NVFlare."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6fda7082-767b-48fb-b73e-2b75b320bf53",
+   "metadata": {},
+   "source": [
+    "## Building a Docker Image\n",
+    "\n",
+    "### Building a Docker Image with a Dockerfile\n",
+    "At the root of the NVFlare repository, there is a docker directory with a Dockerfile inside:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "92746594-b27c-4c89-80ef-17256c533b93",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! cd ../../.. && cat docker/Dockerfile"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "87aba3a0-e65a-49f7-9f27-7975bc335079",
+   "metadata": {},
+   "source": [
+    "Note that this Dockerfile uses a base container of python3.8 and then copies files from the root of the NVFlare repository while updating and installing the most basic dependencies.\n",
+    "\n",
+    "To build a Docker image named `nvflare-docker` with this Dockerfile, you can use the following command:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "05f41f3a-03f8-42e4-882f-fee4d5e8342f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! cd ../../.. && docker build -t nvflare-docker . -f docker/Dockerfile"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "486a18f9-650d-4b35-bbdc-52ba07761d51",
+   "metadata": {},
+   "source": [
+    "You can check that the Docker image has been built and exists with:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "740bb26e-87f5-4561-9384-189c0c423822",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! docker images"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9094f2cc-3463-4b8d-a347-50556d413423",
+   "metadata": {},
+   "source": [
+    "In this example Dockerfile, we are using the python3.8 base container. You can use other base containers if you need other dependencies.\n",
+    "\n",
+    "### Dependencies of Base Images\n",
+    "Please note that there may be issues if the dependencies of the base image you are using have versions that conflict with NVIDIA FLARE. For example, the PyTorch base image also has gRPC, so if the version of gRPC in the PyTorch base image is not compatible with the version used by NVIDIA FLARE, you may run into errors and not be able to successfully connect the FL clients and servers."
```

Comment:
I think we don't need to provide an example here.

If we really want to provide an example, we need to be specific about "the PyTorch base image"
What we are referring to is this: https://catalog.ngc.nvidia.com/orgs/nvidia/containers/pytorch
otherwise, other pytorch image may be good.